students/johnpharmd/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+# started with form_st = 'file_{:d}: {:f}, {:d}, {:f}'
+# Internet search pointed me to https://docs.python.org/2/library/string.html,
+# which, apparently, is not (completely) deprecated in Python 3.
+# Also used https://docs.python.org/3.3/library/string.html#formatstrings.
+# iteration: form_st = 'file_{:03d}: {:f}, {:e}, {:e}'
+# iteration: form_st = 'file_{:03d}: {:.2f}, {:e}, {:e}'
+# iteration: form_st = 'file_{:03d}: {:.2f}, {:.2e}, {:e}'
+# Task 1
+ form_st = 'file_{:03d}: {:.2f}, {:.2e}, {:.2e}'
+
+ # Task 2
+ # started with:
+ # form_st2 = 'file_{integer}: {flo}, {expo1}, {expo2}'
+ # Found this:
+ # https://stackoverflow.com/questions/3228865/how-do-i-format-a-number-with-a-variable-number-of-digits-in-python
+ form_st1 = 'file_{num:0{width}}: {flo:.2f}, {expo1:.2e}, {expo2:.2e}'
+
+
+# Task 3
+# Rewrite 'the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3)' to
+# take an arbitrary number of values.
+
+# form_string = 'the {} numbers are:...'
+# see Chris B's video at 5:10 in
+def formatter(in_tuple):
+    l = len(in_tuple)
+    output0 = 'the {} numbers are: '
+    output1 = ', '.join(['{}'] * l)
```

Comment:
You could use in place assignment here so you don't need the second `output` variable:

`output += ', '.join(['{}'] * len(in_tuple))`